datasource/opennms-kafka/src/main/java/org/opennms/oce/datasource/opennms/AlarmToInventory.java

```diff
@@ -28,133 +28,41 @@
+    private final ScriptedInventoryService inventoryService;
 
-        final InventoryModelProtos.InventoryObjects.Builder iosBuilder = InventoryModelProtos.InventoryObjects
-                .newBuilder();
-        final InventoryModelProtos.InventoryObjects ios;
-        if (!Strings.isNullOrEmpty(alarm.getManagedObjectType()) &&
-                !Strings.isNullOrEmpty(alarm.getManagedObjectInstance())) {
-            final InventoryFromAlarm inventoryFromAlarm = getInventoryFromAlarm(alarm);
-            for (InventoryModelProtos.InventoryObject io : inventoryFromAlarm.getInventory()) {
-                iosBuilder.addInventoryObject(io);
-            }
-            ios = iosBuilder.build();
-            if (inventoryFromAlarm.getManagedObjectInstance() != null && inventoryFromAlarm.getManagedObjectType() != null) {
-                managedObjectInstance = inventoryFromAlarm.getManagedObjectInstance();
-                managedObjectType = inventoryFromAlarm.getManagedObjectType();
-            } else if (ios.getInventoryObjectCount() > 0) {
-                final InventoryModelProtos.InventoryObject io = ios.getInventoryObject(0);
-                managedObjectInstance = io.getId();
-                managedObjectType = io.getType();
-            }
-        } else {
-            ios = iosBuilder.build();
-        }
-
-        if ((managedObjectInstance == null  || managedObjectType == null) && alarm.hasNodeCriteria()) {
-            final String nodeCriteria = OpennmsMapper.toNodeCriteria(alarm.getNodeCriteria());
-            managedObjectType = ManagedObjectType.Node.getName();
-            managedObjectInstance = nodeCriteria;
-        }
-
-        return new EnrichedAlarm(alarm, ios, managedObjectType, managedObjectInstance);
+    public AlarmToInventory(ScriptedInventoryService inventoryService) {
+        this.inventoryService = inventoryService;
     }
 
-    private static final class InventoryFromAlarm {
-
-        private final List<InventoryModelProtos.InventoryObject> inventory;
-        private final String managedObjectInstance;
-        private final String managedObjectType;
-
-        public InventoryFromAlarm(List<InventoryModelProtos.InventoryObject> inventory) {
-            this(inventory, null, null);
-        }
-
-        public InventoryFromAlarm(List<InventoryModelProtos.InventoryObject> inventory, String managedObjectInstance, String managedObjectType) {
-            this.inventory = Objects.requireNonNull(inventory);
-            this.managedObjectInstance = managedObjectInstance;
-            this.managedObjectType = managedObjectType;
-        }
-
-        public List<InventoryModelProtos.InventoryObject> getInventory() {
-            return inventory;
-        }
-
-        public String getManagedObjectInstance() {
-            return managedObjectInstance;
-        }
-
-        public String getManagedObjectType() {
-            return managedObjectType;
-        }
+    public void init() {
```

Comment:
This probably doesn't need an `init` either, and the ones in the rest of the *toInventory classes.

Reply:
removed